.circleci/config.yml

```diff
@@ -0,0 +1,112 @@
+version: 2.1
+
+references:
+  images:
+    ruby: &RUBY_IMAGE circleci/ruby:2.3.4-node
+
+aliases:
+  - &restore_gem_cache
+      keys: 
+        - gemfile-{{ checksum "Gemfile.lock" }}
+  - &save_gem_cache
+      name: Save gem cache
+      key: gemfile-{{ checksum "Gemfile.lock" }}
+      paths: 
+        - ~/data/vendor/bundle
+  - &attach_workspace
+      attach_workspace:
+        at: ~/data
+
+commands:
+  bundle-install:
+    description: Bundle Install
+    parameters:
+      bundle-version:
+        type: string
+      gem-version:
+        type: string
+    steps:
+      - run: 
+          name: Upgrade bundler
+          command: |
+            gem install bundler -v '~> << parameters.bundle-version >>'
+            bundle '_<< parameters.bundle-version >>_' install
+      - run: 
+          name: Upgrade gem
+          command: |
+            sudo gem update --system '<< parameters.gem-version >>'
+      - run: 
+          name: Bundle install
+          command: |
+            bundle install --jobs 7 --retry=3 --path=vendor/bundle
+      - run: 
+          name: Bundle exec rake
+          command: |
+            bundle exec rake
+
+jobs:
+  build:
+    docker:
+      - image: *RUBY_IMAGE
+    working_directory: ~/data
+    steps:
+      - checkout
+      - *attach_workspace
+      - restore_cache: *restore_gem_cache
+      - bundle-install:
+          bundle-version: '1.17.3'
+          gem-version: '3.1.2'
+      - save_cache: *save_gem_cache
+      - persist_to_workspace:
+          root: .
+          paths:
+            - vendor/bundle
+  gem-push:
+    docker:
+      - image: *RUBY_IMAGE
+    working_directory: ~/data
```

Comment:
Do you need to set a `working_directory` here? Or can it just use the default?

Reply:
I changed this around a bit after testing. Calling `bundle exec rake release` from ~/data works without needing any other commands, when the contents of ~/data/vendor/bundle are saved/restored from the previous step

Reply:
Let me know how this looks now, and thanks again for the review, Alvin!